include/seqan3/io/record.hpp

```diff
@@ -232,9 +239,12 @@ struct record : detail::transfer_template_args_onto_t<field_types, std::tuple>
                   "You must give as many IDs as types to seqan3::record.");
 
     //!\brief Clears containers that provide `.clear()` and (re-)initialises all other elements with `= {}`.
-    void clear() noexcept(noexcept(std::apply(expander, std::declval<record &>())))
+    void clear() noexcept(noexcept(std::apply(expander, std::declval<record &>().as_base())))
     {
-        std::apply(expander, *this);
+        // PR2165 / __cpp_lib_tuple_like (C++23): std::apply requires tuple-like.
+        // In C++23, this means std::array, std::pair, std::tuple, and std::ranges::subranges. Nothing else.
+        // https://en.cppreference.com/w/cpp/utility/tuple/tuple-like
+        std::apply(expander, as_base());
```

Comment:
Could also be
```cpp
std::apply(expander, static_cast<base_type &>(*this));
```

but I prefer the `as_base()`

Reply:
Agree, `as_base()` is nicer